lib/teiserver/lobby/libs/lobby_restrictions.ex

```diff
@@ -0,0 +1,282 @@
+          msg = get_failed_rank_check_text(user.rank, state)
+          {:error, msg}
+
+        state.maximum_rank_to_play != nil and user.rank > state.maximum_rank_to_play ->
+          # Send message
+          msg = get_failed_rank_check_text(user.rank, state)
+          {:error, msg}
+
+        true ->
+          {:ok, nil}
+      end
+    end
+  end
+
+  @doc """
+  Returns {:ok, nil} or {:error,msg}
+  """
+  def check_rating_to_play(user_id, consul_state) do
+    state = consul_state
+    team_size = state.host_teamsize
+    team_count = state.host_teamcount
+
+    rating_type =
+      cond do
+        team_count > 2 && team_size == 1 -> "FFA"
+        team_count > 2 && team_size > 1 -> "Team FFA"
+        team_size == 1 -> "Duel"
+        true -> "Team"
+      end
+
+    {player_rating, _player_uncertainty} =
+      BalanceLib.get_user_rating_value_uncertainty_pair(user_id, rating_type)
+
+    cond do
+      state.minimum_rating_to_play != nil and player_rating < state.minimum_rating_to_play ->
+        msg = get_failed_rating_check_text(player_rating, state, rating_type)
+        {:error, msg}
+
+      state.maximum_rating_to_play != nil and player_rating > state.maximum_rating_to_play ->
+        msg = get_failed_rating_check_text(player_rating, state, rating_type)
+        {:error, msg}
+
+      true ->
+        # All good
+        {:ok, nil}
+    end
+  end
+
+  @doc """
+  You cannot have all welcome lobby name if there are restrictions
+  Returns {:ok, nil}
+  Or {:error, msg}
+  """
+  def check_lobby_name(name, consul_state) do
+    cond do
+      has_restrictions?(consul_state) and allwelcome_name?(name) ->
+        {:error,
+         "* You cannot declare a lobby to be all welcome if there are player restrictions"}
+
+      is_noob_title?(name) ->
+        {:ok,  get_noob_looby_tips()}
+
+      true ->
+        {:ok, nil}
+    end
+  end
+
+  defp get_noob_looby_tips() do
+    [
+      @splitter, "Noob lobby tips", @splitter,
+      "To restrict this lobby to players who are new, use command:",
+      "$maxchevlevel <chevlevel>",
+      "To ensure new players are distributed evenly across teams, use command:",
+      "$balancemode split_one_chevs"
+    ]
+  end
+
+  @doc """
+  Check if lobby has restrictions for playing
+  """
+  defp has_restrictions?(consul_state) do
+    state = consul_state
+
+    cond do
+      state.maximum_rating_to_play < @rating_upper_bound -> true
+      state.minimum_rating_to_play > 0 -> true
+      state.minimum_rank_to_play > 0 -> true
+      state.maximum_rank_to_play < @rank_upper_bound -> true
+      true -> false
+    end
+  end
+
+  # Teifion added code to prevent setting restrictions to All Welcome lobbies
+  def allowed_to_set_restrictions(state) do
+    name =
+      state.lobby_id
+      |> Battle.get_lobby()
+      |> Map.get(:name)
+
+    cond do
+      allwelcome_name?(name) -> {:error, "You cannot set a rating limit if all are welcome to the game"}
+      true -> {:ok, nil}
+    end
```

Comment:
:D you do really like the `cond` structure. `if` or `unless` would do the job just fine here :D